src/io/perun/core.clj

```diff
@@ -11,15 +12,15 @@
   "Return metadata on files. Files metadata is a list.
    Internally it's stored as a map indexed by `:path`"
   [fileset]
-  (-> fileset meta +meta-key+ vals))
+  (keep +meta-key+ (vals (:tree fileset))))
 
 (defn key-meta [data]
   (into {} (for [d data] [(:path d) d])))
 
 (defn set-meta
-  "Update `+meta-key+` metadata for the fileset and return updates fileset"
+  "Update `+meta-key+` metadata for files in `data` and return updated fileset"
   [fileset data]
-  (vary-meta fileset assoc +meta-key+ (key-meta data)))
+  (boot/add-meta fileset (into {} (for [d data] [(:path d) {+meta-key+ d}]))))
```

Comment:
I had to look it up, I am pretty new to boot.  Here is the commit that added it https://github.com/boot-clj/boot/commit/1ebcbb8dd78ee4fb88dad70e629630f9442a8b7a.  I think this means it has been in boot since 2.0?